src/restic/hardlinks_index.go

```diff
@@ -0,0 +1,99 @@
+package restic
+
+import (
+	"sync"
+	"fmt"
+)
+
+// HardlinkKey is a composed key for finding inodes on a specific device
+type HardlinkKey struct {
+    Inode, Device uint64
+}
+
+// HardlinkData contains to number of links for an inode, and the filename
+type HardlinkData struct {
+    Links uint64
+    Name string
+}
+
+var (
+	hardLinkIndex      = make(map[HardlinkKey]*HardlinkData)
```

Comment:
Having just on global instance is a bad idea, make this a separate type and embed the mutex. Also, please just use a simple `Mutex`, not a `sync.RWMutex`, that's not really needed here.